File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        st=0
        ans=0
        end=len(height)-1
        while(st<end):
            f=height[st]
            s=height[end]
            m=min(f,s)
            dist=end-st
            area=m*dist
            ans=max(area,ans)
            if f<s:
                st+=1
            else:
                end-=1
        return ans

        

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int st = 0;
        int end = height.length-1;
        int max_area = 0;
        // int area = 0;
        while(st<end){
           max_area = Math.max(max_area,((Math.min(height[st],height[end])) * (end-st)));
           if(height[st]<height[end]){
            st++;
           }
           else{
            end--;
           }
        }
        return max_area;
    }
}